packages/headless/src/controllers/url-manager/headless-url-manager.ts

```diff
@@ -45,6 +45,13 @@ export interface UrlManager extends Controller {
    * @param fragment The part of the url that contains search parameters.  E.g., `q=windmill&f[author]=Cervantes`
    */
   synchronize(fragment: string): void;
+  /**
+   * Checks whether the specified fragment matches the value in state.
+   *
+   * @param fragment The part of the url that contains search parameters.  E.g., `q=windmill&f[author]=Cervantes`
+   * @returns `true` if the passed fragment matches the value in state, and `false` otherwise.
+   */
+  hasFragment(fragment: string): boolean;
```

Comment:
Utility method to compare fragments, because its often encoded, implementers won't always think of decoding before comparing (I didn't 😅 )

Reply:
Following on the earlier comment, is this public method needed if using `pushState` to set the hash?